src/channel.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * Default Channel handling Real Web App
+ */
+
+import { PermissionDataType } from "./user"
+
+export enum IframeActionKind {
+	RequestPermission = "requestPermission",
+	ShowPrompt = "showPrompt",
+	GetUserDetails = "getUserDetails",
+	LoadExperience = "loadExperience",
+}
+
+export type IframeAction<T> = {
+	type: IframeActionKind
+	payload?: { data: T }
+}
+
+// EVENTS
+/** To Send Data from experience to real app */
+export const postToRealAppEvent = (action: IframeAction<any>): void => {
+	// Send Post message to any available window method. As web and native have their respective listeners
+
+	window?.parent?.postMessage(action, "*")
+	;(window as any)?.ReactNativeWebView?.postMessage(JSON.stringify(action), "*")
```

Comment:
This function is being called from the experience, so that exp/website doesn't know if the experience is being run in Native webView or Web's iframe.
Here we're Sending 2 postMessages and based on the platform one or the other is being listened. and the other does nothing.
Other ways might become very complicated to detect the platform and do a conditional call based on that